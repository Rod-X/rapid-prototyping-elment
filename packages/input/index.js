import RodxInput from './src/input.vue'

RodxInput.install = Vue => {
  Vue.component(RodxInput.name, RodxInput)
}

export default RodxInput
