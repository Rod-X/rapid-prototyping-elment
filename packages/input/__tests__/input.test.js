import input from '../src/input.vue'
import {
    mount
} from "@vue/test-utils";

desctipt('rod-input', () => {

    test('input-text', () => {
        const wrapper = mount(input)
        expect(wrapper.html()).toContain('input typr="text"')
    })

    test('input-password',()=>{
        const wrapper = mount(input,{
            propsData:{
                type:'password'
            }
        })
        expect(wrapper.html()).toContain('input type="password')
    })
    
    test('input-value',()=>{
        const wrapper = mount(input,{
            propsData:{
                type:'password',
                value:'myInput'
            }
        })
        expect(wrapper.html()).toContain('myInput')
    })

    test('input-snapshot',()=>{
        const wrapper = mount(input,{
            type:'text',
            value:'myInput'
        })
        expect(wrapper.vm.$el).toMatchSnapshot()
    })
})