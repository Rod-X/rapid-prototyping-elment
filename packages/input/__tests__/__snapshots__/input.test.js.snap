// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rod-input input-snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
